apps/builder/app/services/token.server.ts

```diff
@@ -0,0 +1,178 @@
+import jwt from "@tsndr/cloudflare-worker-jwt";
+import { z } from "zod";
+
+// Ensure your key is of the correct length
+const normalizeKey = (key: string, length = 32) => {
+  const encoder = new TextEncoder();
+  const keyBuffer = encoder.encode(key);
+
+  if (keyBuffer.length === length) {
+    return keyBuffer;
+  }
+
+  if (keyBuffer.length > length) {
+    // Truncate the key if it's too long
+    return keyBuffer.slice(0, length);
+  } else {
+    // Pad the key if it's too short
+    const paddedKey = new Uint8Array(length);
+    paddedKey.set(keyBuffer);
+    return paddedKey;
+  }
+};
+
+export const encrypt = async (text: string, key: string) => {
+  const enc = new TextEncoder();
+  const keyBuffer = await crypto.subtle.importKey(
+    "raw",
+    normalizeKey(key),
+    { name: "AES-GCM" },
+    false,
+    ["encrypt"]
+  );
+  const iv = crypto.getRandomValues(new Uint8Array(12));
+  const ciphertext = await crypto.subtle.encrypt(
+    { name: "AES-GCM", iv },
+    keyBuffer,
+    enc.encode(text)
+  );
+
+  return Buffer.from([...iv, ...new Uint8Array(ciphertext)]).toString(
+    "base64url"
+  );
+};
+
+export const decrypt = async (ciphertext: string, key: string) => {
+  const keyBuffer = await crypto.subtle.importKey(
+    "raw",
+    normalizeKey(key),
+    { name: "AES-GCM" },
+    false,
+    ["decrypt"]
+  );
+  const data = Uint8Array.from(Buffer.from(ciphertext, "base64url"));
+
+  const decrypted = await crypto.subtle.decrypt(
+    { name: "AES-GCM", iv: data.slice(0, 12) },
+    keyBuffer,
+    data.slice(12)
+  );
+  return new TextDecoder().decode(decrypted);
+};
+
+export const verifyChallenge = async (
+  codeVerifier: string,
+  codeChallenge: string
+): Promise<boolean> => {
+  const encoder = new TextEncoder();
+  const data = encoder.encode(codeVerifier);
+  const hashBuffer = await crypto.subtle.digest("SHA-256", data);
+  const base64Hash = Buffer.from(hashBuffer).toString("base64url");
+
+  return base64Hash === codeChallenge;
+};
+
+const CodeTokenPayloadEncSchema = z.object({
```

Comment:
we really didn't change the overall convention for zod schema naming, so far we used without suffix

Reply:
searching through codebase I can see now 2 notations ... really not liking this fact

Reply:
if we decided to use Schema prefix then lets rename the other places too